.eslintrc.json

```diff
@@ -2,6 +2,13 @@
   "root": true,
   "ignorePatterns": ["**/*"],
   "plugins": ["@nrwl/nx", "sonarjs"],
+  "parser": "@typescript-eslint/parser",
+  "parserOptions": {
+    "ecmaVersion": 2018,
+    "ecmaFeatures": { "modules": true },
+    "sourceType": "module",
+    "project": ["./tsconfig.base.json", "./apps/**/tsconfig.app.json", "./libs/**/tsconfig.lib.json"]
```

Comment:
That makes sense. I'll exclude the `./tsconfig.base.json`. For the other project, they support glob patterns according to their documentation, so I would expect this to be necessary.

I prefer it this way; otherwise, doing this per project in larger projects would be a pain.

Reply:
When generating a library using the `@nrwl/angular:library` generator with `--setParserOptionsProject=true`, this is what's added to `.eslintrc.json`:

```json
{
  "overrides": {
    "parserOptions": {
        "project": ["libs/my-lib/tsconfig.*?.json"]
      }
  }
}
```

question: Seeing that every project ignores files that are outside of its source directory, I would think that configuring it on the workspace-level should work but how can we verify that this is the case?

suggestion: Based on the generated settings above, we could simplify the `parserOptions.project` setting to:
```json
{
  "overrides": {
    "parserOptions": {
        "project": ["{apps,libs}/**/tsconfig.*?.json"]
      }
  }
}
```

This would also include TypeScript configurations such as `*tsconfig.lib.prod.json` and `tsconfig.*.spec.json` which are currently excluded.

Reply:
> I prefer it this way; otherwise, doing this per project in larger projects would be a pain.

We won't have a ton of projects in our workspaces but it still leaves room for human error so either we use workspace-level settings like you're proposing or we add these configurations to `nx.json`:
```json
{
  "generators": {
     "@nrwl/angular:application": {
      "setParserOptionsProject": true
    }
    "@nrwl/angular:library": {
      "setParserOptionsProject": true
    }
  }
}
```

This only leaves `e2e` projects without a `parserOptions.project` setting.

Reply:
I added a new glob pattern for e2e projects in the global **tsconfig** file.

I like the alternative that you are proposing about forcing the configuration of the generator, but if the global config is working correctly I would accept that for now.

How would the `parserOptions.project`  of the root `tsconfig.json` look like if each library configures its own?

Reply:
> Seeing that every project ignores files that are outside of its source directory, I would think that configuring it on the workspace level should work, but how can we verify that this is the case?

Without the latest adjustments I did for the e2e glob, the pipeline failed because the files belonging to the e2e projects were not included in any project. Isn't that a verification that it is working?

Reply:
> How would the parserOptions.project of the root tsconfig.json look like if each library configures its own?

It wouldn't be configured. This is a base configuration file.

Reply:
> Without the latest adjustments I did for the e2e glob, the pipeline failed because the files belonging to the e2e projects were not included in any project. Isn't that a verification that it is working?

We would need to trigger a lint error for a rule that we know requires type information. It could be from a different ESLint plugin than SonarLint.

Reply:
Found this on the Nx documentation

> and you should not configure parserOptions.project in your workspace's root .eslintrc.json